File: app/src/main/java/ir/ghararemaghzha/game/dialogs/NoInternetDialog.java
package ir.ghararemaghzha.game.dialogs;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.widget.Toast;

import androidx.annotation.NonNull;

import ir.ghararemaghzha.game.R;
import ir.ghararemaghzha.game.classes.RetryInterface;
import ir.ghararemaghzha.game.classes.Utils;

public class NoInternetDialog extends Dialog {
    private final Context context;
    private final RetryInterface retryInterface;

    public NoInternetDialog(@NonNull Context context, RetryInterface retryInterface) {
        super(context);
        this.context = context;
        this.retryInterface = retryInterface;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_nointernet);
        findViewById(R.id.alert_dialog_set).setOnClickListener(v -> {
            if (Utils.checkInternet(context)) {
                dismiss();
                retryInterface.retry();
            } else {
                Toast.makeText(context, R.string.general_internet_error, Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: app/src/main/java/ir/ghararemaghzha/game/dialogs/NoInternetDialog.kt
package ir.ghararemaghzha.game.dialogs

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.widget.Toast
import ir.ghararemaghzha.game.R
import ir.ghararemaghzha.game.classes.Utils
import ir.ghararemaghzha.game.databinding.DialogNointernetBinding

class NoInternetDialog(ctx: Context,private val retry:()->Unit) : Dialog(ctx) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val b = DialogNointernetBinding.inflate(LayoutInflater.from(context))
        setContentView(b.root)
       b.alertDialogSet.setOnClickListener {
            if (Utils.checkInternet(context)) {
                retry()
                dismiss()
            } else {
                Toast.makeText(context, R.string.general_internet_error, Toast.LENGTH_SHORT).show()
            }
        }
    }
}
